Порівняння двох версій рекламної кампанії:

кампанія A була показана 2000 користувачам, і 220 з них здійснили покупку;\
кампанія B була показана 2500 користувачам, і 240 з них здійснили покупку.

Виконаємо тест пропорцій, щоб перевірити, чи існує статистично значуща різниця між кампаніями.

In [ ]:
import statsmodels.api as sm


# data of views and purchases
views_a = 2000
views_b = 2500
purchases_a = 220
purchases_b = 240


# perform z-test for two proportions
z_stat, p_value = sm.stats.proportions_ztest([purchases_a, purchases_b], [views_a, views_b])

print(f"Z-statistic: {z_stat}")
print(f"p-value: {p_value}")

# Interpretation
if p_value < 0.05:
  print("There is a statistically significant difference between versions A and B.")
else:
  print("No statistically significant difference between versions A and B.")

Z-statistic: 1.540457412600049
p-value: 0.12344889604606361
No statistically significant difference between versions A and B.


**Висновок:**\
За результатами Z-тесту (p-значення = 0.1234 > 0.05) немає статистично значущої різниці у конверсії між рекламними кампаніями A та B. Це означає, що кампанії мають подібну ефективність щодо кількості покупок.

Таким чином, на основі наявних даних не виявлено статистично значущої різниці між кампаніями, що свідчить про однакову ефективність обох версій реклами у контексті залучення покупців.